File: src/08-SegregatedInterface.cs
namespace SolidPrinciples.SegregatedInterface
{
    public interface WithdrawAccount
    {
        void Withdraw(decimal amount);
    }

	public interface DepositAccount
	{
		void Deposit(decimal amount);
	}

    public class AccountBase : WithdrawAccount, DepositAccount
    {
        private decimal _overdraft;
        private decimal _balance;

        public AccountBase(decimal overdraft)
        {
            _overdraft = overdraft;
        }

        public void Deposit(decimal amount)
        {
            _balance += amount;
        }

        public void Withdraw(decimal amount)
        {
            _balance += amount;
        }
    }

	public class ATM
	{
		private WithdrawAccount _account;
		
		public ATM (WithdrawAccount account)
		{
			_account = account;
		}

		public void WithdrawOperation(decimal amount)
		{
			_account.Withdraw(amount);
		}
	}
}

File: python/08-segregated-interface.py
class AccountBase:
    def __init__(self, overdraft):
        self._overdraft = overdraft
        self._balance = 0.0

    def deposit(self, amount):
        self._balance += amount

    def withdraw(self, amount):
        self._balance -= amount

class ATM:
    def __init__(self, account):
        self._account = account

    def withdraw_operation(self, amount):
        self._account.withdraw(amount)
